clients/cobol-lsp-vscode-extension/src/commands/RenumCommand.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2025 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import * as vscode from "vscode";
+
+export enum RenumberAction {
+  LEFT,
+  RIGHT,
+}
+
+const maxLines = 999999;
+
+const leftStart = 0;
+const leftEnd = 6;
+const leftDigits = 6;
+const leftMultiplier = 100;
+
+const rightStart = 72;
+const rightEnd = 80;
+const rightMultiplier = 1000;
+const rightDigits = 8;
+/**
+ * Renumber 1-7 or 73-80 columns in active editor.
+ *
+ * @param actionType RenumberAction
+ */
+export async function renumberLines(
+  actionType: RenumberAction,
+  activeEditor: vscode.TextEditor | undefined = vscode.window.activeTextEditor,
+) {
+  if (!activeEditor) {
+    return;
+  }
+  await notificationHandler(
+    renum,
+    activeEditor,
+    actionType,
+    "Renumbering Sequential Lines...",
+  );
+}
+/**
+ * Remove sequential numbers at 1-7 or 73-80 columns in active editor.
+ *
+ * @param actionType RenumberAction
+ */
+export async function unNumberLines(
+  actionType: RenumberAction,
+  activeEditor: vscode.TextEditor | undefined = vscode.window.activeTextEditor,
+) {
+  if (!activeEditor) {
+    return;
+  }
+
+  await notificationHandler(
+    unNumber,
+    activeEditor,
+    actionType,
+    "Removing Sequential Lines...",
+  );
+}
+
+async function unNumber(
+  textEditor: vscode.TextEditor,
+  actionType: RenumberAction,
+) {
+  const { start, end } =
+    actionType == RenumberAction.LEFT
+      ? {
+          start: leftStart,
+          end: leftEnd,
+        }
+      : {
+          start: rightStart,
+          end: rightEnd,
+        };
+
+  await textEditor.edit((editBuilder) => {
+    for (let i = 0; i < textEditor.document.lineCount; i++) {
+      if (textEditor.document.lineAt(i).text.charAt(start) != "*") {
```

Comment:
why does this matter?

Reply:
to ignore the commented out lines

Reply:
yes, but why would you want to ignore them?

Reply:
@VitGottwald at columns 1-6 , there are preprocessor marks like this `*Name:`  should they also considered as same ?

Reply:
Sounds like a bigger question - what if either 1-6 (or 73-80) is neither numeric nor spaces? In that case I suggest we abort the whole command with an error and tell the user the first line that has non-numeric content and what it is, ideally with a link to navigate to it.